### PR TITLE
Fixed check for URI based OS image attributes

### LINKED ISSUE #86: on `compute image update`, privacy uri may change, suggesting failure
```
$ azurectl --account publish compute image --name test-attributes update --privacy-uri http://privacy.uri
INFO: Updating image metadata for: test-attributes
INFO: --> privacy_uri = http://privacy.uri
ERROR: AzureOsImageUpdateError: 'The element(s) "privacy_uri" could not be updated. Please check if your account is registered as image publisher'
```
Yet, viewing the data shows that the attribute *is* assigned, just modified:
```
$ azurectl --account publish compute image --name test-attributes show
{
  "image": {
    ...
    "privacy_uri": "http://privacy.uri/",
    ...
  }
}
```

In the above example, the API appended an ending '/', causing it to fail `azurectl`'s validation.

### PR DESCRIPTION
**Pull request by schaefi**
After an update of e.g privacy_uri the API might have changed the value to be still correct but different from the original value provided on the commandline by e.g a slash (/). Fixes #86

### CODE REVIEW

**Comment by schaefi** [comment]
Sorry for addressing two issues in the same pull request but this is because I could not separate them easily without creating conflicts

**Comment by schaefi** [comment]
you are right, we should be more user friendly :smile: how about it now ?

**Comment by bear454** [comment]
Awesome, Schaefi!

File: azurectl/image.py [hunk 1]
```diff
@@ -255,3 +220,53 @@ def publish(self, name, permission):
...
+            'os': image.os,
+            'os_state': image.os_state,
+            'pricing_detail_link': image.pricing_detail_link,
+            'privacy_uri': image.privacy_uri,
+            'published_date': image.published_date,
+            'publisher_name': image.publisher_name,
+            'recommended_vm_size': image.recommended_vm_size,
+            'show_in_gui': image.show_in_gui,
+            'small_icon_uri': image.small_icon_uri
+        }
```
Comment (rjschwei) [inline azurectl/image.py]: While this is nice and explicit it is also very long and I am not certain that there is value in listing all these attributes one line at a time. How about this?

decorator = []
for image_attr in dir(image):
    if '__' in image_attr:
        continue
    value = getattr(image, image_attr)
    if inspect.ismethod(value):
       continue
   decorator[image_attr] = value

The advantage is that we do not have to list all these attributes explicitly and when a new attribute shows up it goes into the decorator automatically. The disadvantage is that is a new attribute on image is needed that is not supposed to be in the decorator it has to start with "__" of course that test could easily be modified to also include some oher prefix.

**Pull request merged**